packages/freezed/test/common_types_test.dart

```diff
@@ -0,0 +1,122 @@
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:build_test/build_test.dart';
+import 'package:expect_error/expect_error.dart';
+import 'package:test/test.dart';
+
+import 'integration/common_types.dart';
+
+Future<void> main() async {
+  final library = await Library.parseFromStacktrace();
+
+  test('has no issue', () async {
+    final main = await resolveSources(
+      {
+        'freezed|test/integration/common_types.dart': useAssetReader,
+      },
+      (r) => r.libraries.firstWhere(
+          (element) => element.source.toString().contains('common_types')),
+    );
+
+    final errorResult = await main.session
+            .getErrors('/freezed/test/integration/common_types.freezed.dart')
+        as ErrorsResult;
+
+    expect(errorResult.errors, isEmpty);
+  });
+
+  group('CommonSuperSubtype', () {
+    test('has the common properties available', () {
+      var value = const CommonSuperSubtype('a', 1337, 42.24);
+      expect(value.a, 'a');
```

Comment:
This should test that the public type of those properties is respectively `String?`, `num` and `num`

Technically this test could expose those properties as `Object?`, and the test would still pass